apps/api-journeys/src/app/modules/event/video/video.resolver.ts

```
@@ -19,123 +20,336 @@ import {
   }
 }
 @Resolver('VideoPlayEvent')
 @UseGuards(GqlAuthGuard)
 export class VideoPlayEventResolver {
-  constructor(private readonly eventService: EventService) {}
+  constructor(
+    private readonly eventService: EventService,
+    private readonly blockService: BlockService,
+    private readonly visitorService: VisitorService
+  ) {}
+
   @Mutation()
   async videoPlayEventCreate(
     @CurrentUserId() userId: string,
     @Args('input') input: VideoPlayEventCreateInput
   ): Promise<VideoPlayEvent> {
+    const block: { journeyId: string } = await this.blockService.get(
+      input.blockId
+    )
+    const journeyId = block.journeyId
+
+    const visitor = await this.visitorService.getByUserIdAndJourneyId(
+      userId,
+      journeyId
+    )
+
+    const validStep = await this.blockService.validateBlock(
+      input.stepId ?? null,
+      'journeyId',
+      journeyId
+    )
+
+    if (!validStep) {
+      throw new UserInputError(
+        `Step ID ${
+          input.stepId as string
+        } does not exist on Journey with ID ${journeyId}`
+      )
+    }
+
     return await this.eventService.save({
       ...input,
       __typename: 'VideoPlayEvent',
-      userId,
-      createdAt: new Date().toISOString()
+      visitorId: visitor.id,
+      createdAt: new Date().toISOString(),
+      journeyId
     })
   }
 }
 @Resolver('VideoPauseEvent')
 @UseGuards(GqlAuthGuard)
 export class VideoPuaseEventResolver {
-  constructor(private readonly eventService: EventService) {}
+  constructor(
+    private readonly eventService: EventService,
+    private readonly blockService: BlockService,
+    private readonly visitorService: VisitorService
+  ) {}
+
   @Mutation()
   async videoPauseEventCreate(
     @CurrentUserId() userId: string,
     @Args('input') input: VideoPauseEventCreateInput
   ): Promise<VideoPauseEvent> {
+    const block: { journeyId: string } = await this.blockService.get(
+      input.blockId
+    )
+    const journeyId = block.journeyId
```

need to throw a UserInput Error here if no block.